Backend/src/main/java/com/harmonia/backend/controller/UserController.java

```diff
@@ -9,18 +9,38 @@
 import org.springframework.http.HttpStatus;
 import org.springframework.http.ResponseEntity;
 import org.springframework.web.bind.annotation.*;
+
 import java.util.Map;
 
+/**
+ * Rest controller for handling user-related HTTP requests
+ */
+
 @RestController
 @RequestMapping("user")
 public class UserController {
+
+    /**
+     * The service for user-related business logic
+     */
     @Autowired
     private UserService userService;
 
+    /**
+     * Constructor for UserController class
+     *
+     * @param userService the user service to be used
+     */
     public UserController(UserService userService) {
         this.userService = userService;
     }
 
+    /**
+     * Get a list of all users or a list of users by username if a username is provided
+     *
+     * @param userName (optional) the username to search for
+     * @return an iterable list of UserResponse objects
+     */
     @GetMapping
     @CrossOrigin
     public Iterable<UserResponse> listUsers(@RequestParam(name = "username", required = false) String userName) {
```

Comment:
You already have @GetMapping("username/{username}") a few lines above, remove @RequestParam(name = "username", required = false) String userName